src/reporter/index.js

```diff
@@ -168,90 +184,134 @@ export default class Reporter {
         });
     }
 
-    _assignTaskEventHandlers () {
-        const task = this.task;
+    async _onceTaskStartHandler () {
+        const startTime  = new Date();
+        const userAgents = this.task.browserConnectionGroups.map(group => group[0].userAgent);
+        const first      = this.reportQueue[0];
+        const taskProperties = {
+            configuration: this.task.opts
+        };
 
-        task.once('start', async () => {
-            const startTime  = new Date();
-            const userAgents = task.browserConnectionGroups.map(group => group[0].userAgent);
-            const first      = this.reportQueue[0];
-            const taskProperties = {
-                configuration: task.opts
-            };
+        try {
+            await this.plugin.reportTaskStart(
+                startTime,
+                userAgents,
+                this.testCount,
+                this.task.testStructure,
+                taskProperties
+            );
 
-            await this.plugin.reportTaskStart(startTime, userAgents, this.testCount, task.testStructure, taskProperties);
             await this.plugin.reportFixtureStart(first.fixture.name, first.fixture.path, first.fixture.meta);
-        });
+        }
+        catch (e) {
```

Comment:
Since we use the same error everywhere, I think we can generalize the code somehow. So you won't have to spread catches everywhere.